src/main/java/cork/Engine.java

```diff
@@ -0,0 +1,170 @@
+                    displayGameMenu();
+                    break;
+                case GAME_RUNNING:
+                    runGame();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        uiHandler.exit();
+    }
+
+    private void changeState(final String choice)
+    {
+        if (choice.equalsIgnoreCase(QUIT_COMMAND)) {
+            state = State.QUIT;
+        } else {
+            switch (state)
+            {
+                case HOME_SCREEN:
+                    state = State.MAIN_MENU;
+                    break;
+                case MAIN_MENU:
+                    gameName = choice;
+                    state = State.GAME_MENU;
+                    break;
+                case GAME_MENU:
+                    if (choice.equals(UIHandler.LOAD_GAME_OPTION)) { loadGame(); }
+                    state = State.GAME_RUNNING;
+                    break;
+                case GAME_RUNNING:
+                    state = State.HOME_SCREEN;
+                    break;
+                default:
+                    state = State.QUIT;
+                    break;
+            }
+        }
+    }
+
+    private void displayHomeScreen()
+    {
+        changeState(uiHandler.displaySplashScreen());
+    }
+
+    private void displayMainMenu()
+    {
+        try { changeState(uiHandler.displayMainMenu(loadGameList())); }
+        catch (IOException e) {
+            state = State.QUIT;
+            uiHandler.displayError("Please ensure that the Cork JAR is in the same directory as the 'games' folder and that the folder is populated.");
+        }
+    }
+
+    private void displayGameMenu()
+    {
+        currentGame = new Game(gameName);
+        changeState(uiHandler.displayGameMenu(gameName));
+    }
+
+    private List<String> loadGameList() throws IOException {
+        try (Stream<Path> paths = Files.find(Paths.get(GAMES_DIRECTORY_PATH), 1, (path, attributes) -> attributes.isDirectory())) {
+            List<String> gameList = paths.map(Path::getFileName).map(Path::toString).collect(Collectors.toList());
+            gameList.remove(0);
+
+            if (gameList.isEmpty()) { throw new IOException("Games folder empty."); }
+
+            return gameList;
+        }
+    }
+
+    private Path getSaveFilePath()
+    {
+        return Paths.get(String.format("%s/%s/%s", GAMES_DIRECTORY_PATH, gameName, SAVE_FILE_NAME));
+    }
+
+    private void loadGame() {
+        try {
+            for (final String input : Files.readAllLines(getSaveFilePath())) { currentGame.handleCommand(input); }
+            uiHandler.print("Game loaded.");
+        } catch (IOException e) {
+            uiHandler.displayError("Game could not be loaded.");
+        }
+    }
+
+    private void saveGame() {
+        try {
+            Files.write(getSaveFilePath(), currentGame.getPreviousCommands());
+            uiHandler.print("Game saved. Press any key to continue.");
+            uiHandler.getInput();

```

Comment:
The getInput() method requires Enter to be pressed in order to continue. Either the message or the method used need to be changed